src/lda.jl

```diff
@@ -477,30 +477,43 @@ function fit(::Type{SubspaceLDA}, X::AbstractMatrix{T},
     # (essentially, PCA before LDA)
     Uw, Σw, _ = svd(Hw, full=false)
     keep = Σw .> sqrt(eps(T)) * maximum(Σw)
+    count(keep) > 0 || throw(NullMatrixException("within-class covariance matrix is null"))
```

Comment:
You may want to use  `error` function, it will throw `ErrorException` with message provided at the function call. I do not think we need a new exception type.